M2/Macaulay2/packages/Complexes/ChainComplex.m2

```diff
@@ -472,6 +472,24 @@ texMath Complex := String => C -> (
         )
     )
 
+cover Complex := Complex => C -> (
+    (lo, hi) := concentration C;
+    if lo == hi
+    then complex(cover C_lo, Base => lo)
+    else complex applyValues(C.dd.map, cover))
+
+super Complex := Complex => C -> (
+    (lo, hi) := concentration C;
+    if lo == hi
+    then complex(super C_lo, Base => lo)
+    else complex applyValues(C.dd.map, super))
+
+ambient Complex := Complex => C -> (
+    (lo, hi) := concentration C;
+    if lo == hi
+    then complex(ambient C_lo, Base => lo)
+    else complex applyValues(C.dd.map, ambient))
+
```

Comment:
It seems to me that ``cover``, ``super``, and ``ambient`` would not typically produce well-defined complexes.  Am I missing something?

Reply:
Do you have an example in mind? I think they should be functorial, but maybe I'm missing something.